### ISSUE TITLE
=? doesn't work in generic procs

### ISSUE THREAD

**Issue by markspanbroek** [markspanbroek/questionable#2]
The following doesn't compile:

```nim
proc foo[T](option: ?T) =
  if value =? option:
    echo value
  else:
    echo "none"
```

It complains about `value` being an undeclared identifier.

**Comment by markspanbroek** [comment]
Related Nim issue: https://github.com/nim-lang/Nim/issues/15693

**Comment by markspanbroek** [comment]
Can worked around with a `mixin` statement:

```nim
proc foo[T](option: ?T) =
  if value =? option:
    mixin value # workaround
    echo value
  else:
    echo "none"
```

**Comment by markspanbroek** [comment]
Closed in a2023ae
